File: Java/AR_194.java
import java.util.Scanner;

public class AR_194 {
    

    // Enunciado: 

    // Explicacion: 

    // Complejidad:
    // - Preprocesaod: O(nlogn) ordenar por un algoritmo de divide y venceras como
    //  puede ser QuickSort
    // - Temporal: O(n), recorre el vector de n elementos
    // - Espacial: O(2n+4) = O(n) vector de n pares.
    static int voraz(String carretera) {
        int ret=0;
        int i=0, n=carretera.length();
        char cA[]=carretera.toCharArray();

        while (i<n) {
            if (cA[i]=='.') {
                ret++;
                i+=2;
            }

            i++;
        }
        return ret;
    }

    static void resuelveCaso(Scanner scanner) {
        String carretera=scanner.next();
        System.out.println(voraz(carretera));
    }

    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        int nCase=scanner.nextInt();

        for (int i=0;i<nCase;i++) resuelveCaso(scanner);

        scanner.close();
    }
}

File: C/100-200/AR_194.c
#include <stdio.h>
#include <string.h>

// Enunciado: 

// Explicacion: 

// Complejidad:
// - Preprocesaod: O(nlogn) ordenar por un algoritmo de divide y venceras como
//  puede ser QuickSort
// - Temporal: O(n), recorre el vector de n elementos
// - Espacial: O(2n+4) = O(n) vector de n pares.
int voraz(const char* carretera) {
    int ret=0;
    int i=0, n=strlen(carretera);

    while (i<n) {
        if (carretera[i]=='.') {
            ret++;
            i += 2;
        }

        i++;
    }

    return ret;
}
// Demostracion por reduccion de diferencias
// TODO


void resuelveCaso() {
    char carretera[10001];  
    scanf("%s", carretera);
    printf("%d\n", voraz(carretera));
}

int main() {
    int i;

    int nCase;
    scanf("%d", &nCase);
    for (i=0;i<nCase;i++) resuelveCaso();

    return 0;
}